client/components/category-navigation/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { Button } from '@wordpress/components';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import classnames from 'classnames';
+
+import './style.scss';
+
+type CategoryNavProps = {
+	buttons?: {
+		icon: string;
+		label: string;
+		link: string;
+	}[];
+	children?: React.ReactNode;
+	list: {
+		name: string;
+		label?: string;
+		link: string;
+	}[];
+	selectedCategory: string;
+};
+
+export const CategoryNavigation = ( {
+	buttons,
+	children,
+	list,
+	selectedCategory,
+}: CategoryNavProps ) => {
+	const [ showLeftArrow, setShowLeftArrow ] = useState( false );
+	const [ showRightArrow, setShowRightArrow ] = useState( false );
+	const listRef = useRef< HTMLDivElement | null >( null );
+
+	const checkScrollArrows = () => {
+		const list = listRef.current;
+		if ( ! list ) {
+			return;
+		}
+
+		const { scrollLeft, scrollWidth, clientWidth } = list;
+		setShowLeftArrow( scrollLeft > 0 );
+		setShowRightArrow( Math.ceil( scrollLeft ) < scrollWidth - clientWidth );
+	};
+
+	const scrollTo = ( direction: 'right' | 'left' ) => {
+		if ( ! listRef.current ) {
+			return;
+		}
+
+		let left = listRef.current.clientWidth / 2;
+
+		if ( direction === 'left' ) {
+			left *= -1;
+		}
+
+		listRef.current?.scrollBy( { left, behavior: 'smooth' } );
+	};
+
+	useEffect( () => {
+		checkScrollArrows();
+
+		const list = listRef.current;
+		if ( ! list ) {
+			return;
+		}
+
+		list.addEventListener( 'scroll', checkScrollArrows );
+
+		return () => {
+			list.removeEventListener( 'scroll', checkScrollArrows, { passive: true } );
```

Comment:
```suggestion
		listRef.current.addEventListener( 'scroll', checkScrollArrows, { passive: true } );

		return () => {
			listRef.current.removeEventListener( 'scroll', checkScrollArrows );
```
Referencing `listRef.current` directly is just my personal preference (I see no reason to use the `list` variable), but `{ passive: true }` should definitely be moved from the `removeEventListener` call to `addEventListener`

Reply:
> Referencing listRef.current directly is just my personal preference (I see no reason to use the list variable)

![Screenshot 2024-03-07 at 13 23 23](https://github.com/Automattic/wp-calypso/assets/5598437/89ab3367-90a0-4fa9-9741-11281ca8b4fa)

> but { passive: true } should definitely be moved from the removeEventListener call to addEventListener
Oh, sharp eyes Fredrik, thanks. It's indeed should be there, just typo.